### PR TITLE
Feature: Register new user

### PR DESCRIPTION
**Pull request by chphungphat**
Other changes:
- move authenticate method into UserServiceImpl
- PasswordEncoder move into new class BeanContext to prevent circular injection
- Add regex for email, password, gender, phone number into application.properties
- Add back in sql scripts.
- Add JwtAuthEntryPoint to handle exception
- Change security filter to permit all for all api request for easier testing

### CODE REVIEW

File: src/main/java/com/cg_vibely_social_service/entity/Gender.java
Comment (ChungNguyenVr912) [inline src/main/java/com/cg_vibely_social_service/entity/Gender.java:1]: not entity

File: src/main/java/com/cg_vibely_social_service/dto/converter/UserConverter.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Component;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+@Component
+public class UserConverter {
```
Comment (ChungNguyenVr912) [inline src/main/java/com/cg_vibely_social_service/dto/converter/UserConverter.java]: implement interface for DI

**Review by ChungNguyenVr912 — CHANGES_REQUESTED** [review]

**Review by ChungNguyenVr912 — APPROVED** [review]
look better

**Review by MinhTam1509 — APPROVED** [review]
that's good

**Review by HuyVu18 — APPROVED** [review]

**Pull request merged**